mesonbuild/mesonmain.py

```diff
@@ -144,10 +144,14 @@ def run(self, args):
             if os.environ.get('MESON_FORCE_BACKTRACE'):
                 raise
             return 1
-        except Exception:
+        except Exception as e:
             if os.environ.get('MESON_FORCE_BACKTRACE'):
                 raise
             traceback.print_exc()
+            args = ()
```

Comment:
The typing for this is
```python
Tuple[Union[int, str], ...]
```

Reply:
This file is totally untyped anyway...

But, it's actually a tuple of the form `(string, integer, integer)`.

Reply:
But it's not really, it's either that or an empty tuple. But to satisfy Moog.exception what i wrote is fine and correct. Alternatively you could use a list, which is really the "correct" type here

Reply:
Why is a list the correct type here? Lists are homogeneous, while each element of this variable has specific and different meaning. Changing it to a list would just lead to using unverifiable `List[Union[int, str]]` which, again, doesn't know that `[integer, string, string]` is invalid and not permitted.

The correct typing here is:

```
Union[Tuple[str, int, int], Tuple[()]]
```

And all of this seems quite irrelevant since the file isn't typed to begin with and isn't run under mypy.

Reply:
Well, you've added `# type: ignore` below, so we might as well just do it right the first time. But yes, you could write this as `Tuple[str, int, int] | Tuple[()]`, but the splat (`*`) operator works on iterables, so any iterable is fair game, and `List[str | int]` is technically correct and conceptually simpler. maybe though it would make more sense to write as
```python
msg = 'Unhandled python exception'
if al(...)
    e = MesonBugException(msg, e.file, e.lineno, e.colno)
else:
    e = MesonBugException(msg)
mlog.exception(e)

Reply:
I added that as copypasta from mlog.py :D 

if/else seems like a nice way to sidestep the problem though, and it does indeed seem clearer to read.